transforms/pair.go

```diff
@@ -0,0 +1,137 @@
+package transforms
+
+import (
+	"github.com/facebookgo/errgroup"
+	"gopkg.in/Clever/optimus.v3"
+)
+
+// PairType is the type of join to use when Pairing
+type PairType int
+
+// RowHasher takes in a row and returns a hash for that Row.
```

Comment:
I don't like `RowJoinKey` because this is also used by `Unique`, and could be used by `Sort` in the future if we ever do that, but `RowIdentifier` sounds good to me.